### PR TITLE
fix: allow to resume/see progress of a build image task

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
It is not yet using the task manager but I reworked the process to be able to reconnect to an existing background process

### Screenshot/screencast of this PR

https://user-images.githubusercontent.com/436777/222752149-a23ca285-f7d9-47ea-b314-a18afd481a0f.mp4

### What issues does this PR fix or reference?

#125 

### How to test this PR?

Try to build an image, go elsewhere and come back. It should resume the process

example of Containerfile
```
FROM alpine
RUN sleep 30
RUN echo "end"
```

Change-Id: Iab0b01f3f8cd72b640b2c6ab1b2634e76b42ec4e

### CODE REVIEW

File: packages/main/src/plugin/container-registry.ts [hunk 1]
```diff
@@ -1030,8 +1030,10 @@ export class ContainerProviderRegistry {
     // eslint-disable-next-line @typescript-eslint/ban-types
     function onFinished(err: Error | null, output: {}) {
       if (err) {
+        eventCollect('finish', '' + err);
```
Comment (lstocchi) [inline packages/main/src/plugin/container-registry.ts]: eventCollect expects a string, i think you need to have `err.message` or whatever other prop you want to show.

**Review by lstocchi — COMMENTED** [review]
Worked fine with your example but i think we should add a button to cancel a failing/blocked build. 
I ended up in this situation and i wasn't able to do anything.
i wanted to create a new Dockerfile and i used vscode. It created a <title>.dockerfile file (maybe that was the problem?!). BTW by using it this was the result.

error-build

File: packages/main/src/plugin/container-registry.ts (hunk 1 above)
Comment (benoitf) [reply packages/main/src/plugin/container-registry.ts]: You don't like toString() on the error object ?

Comment (lstocchi) [reply packages/main/src/plugin/container-registry.ts]: The prob is that if it does not overwrite the default nodejs toString i think you would only see [Object object]. I don't know how to make it call the onFinished with an error but if you say it prints the info correctly, i trust you :+1:

Comment (benoitf) [reply packages/main/src/plugin/container-registry.ts]: I can enhance to provide an Error object.

but yes it doesn't provide [Object object]

```bash
$ node -p "new Error('this is an error').toString()"
Error: this is an error
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/toString#using_tostring

**Comment by benoitf** [comment]
@lstocchi I suspect the problem you raised is because it's uploading the build context from a large directory
I've added some traces to let the user know

if you want to be able to abort a build task, probably it needs to be a separate PR

**Review by lstocchi — APPROVED** [review]
LGTM

i vote to have the abort action somewhere as it can be useful imo. 
I agree to have it in a separate PR.

**Comment by benoitf** [comment]
created https://github.com/containers/podman-desktop/issues/1655

**Pull request merged**